pedpy/methods/profile_calculator.py

```diff
@@ -383,57 +472,86 @@ def compute_speed_profile(
+    across a grid within a walkable area.
+    The method of computation can be selected among several options,
+    including mean (:attr:`SpeedMethod.MEAN`),
+    Gaussian (:attr:`SpeedMethod.GAUSSIAN`),
+    Voronoi (:attr:`SpeedMethod.VORONOI`),
+    and arithmetic mean methods (:attr:`SpeedMethod.ARITHMETIC`),
+    each suitable for different analysis contexts.
 
     Args:
-        data: Data from which the profiles are computes.
-            The DataFrame must contain a `frame` and a `speed` (result from
-            :func:`~speed_calculator.compute_individual_speed`) column.
-            For computing density profiles, it must contain a `polygon` column
+        data: A pandas DataFrame containing `frame` and pedestrian `speed`
+            (result from :func:`~speed_calculator.compute_individual_speed`).
+            Depending on `speed_method`, additional columns `x`, `y`, or
+            `polygon` might be required. `polygon` column
             (from :func:`~method_utils.compute_individual_voronoi_polygons`)
-            when using the `DensityMethod.VORONOI`. When computing the classic
-            density profile (`DensityMethod.CLASSIC`) the DataFrame needs to
-            contain the columns 'x' and 'y'. Computing the speed
-            profiles needs a `polygon` column (from
-            :func:`~method_utils.compute_individual_voronoi_polygons`) when
-            using the `SpeedMethod.VORONOI` or `SpeedMethod.ARITHMETIC`.
+            is required when using the :attr:`SpeedMethod.VORONOI` or
+            :attr:`SpeedMethod.ARITHMETIC`.
+            When computing the Gaussian profile (:attr:`SpeedMethod.GAUSSIAN`)
+            the DataFrame needs to contain the columns `x` and `y`.
             For getting a DataFrame containing all the needed data, you can
-            merge the results of the different function on the 'id' and
-            'frame' columns (see :meth:`pandas.DataFrame.merge` and
+            merge the results of the different function on the `id` and
+            `frame` columns (see :func:`pandas.DataFrame.merge` and
             :func:`pandas.merge`).
-        walkable_area (WalkableArea): geometry for which the profiles are
-            computed
-        grid_size: resolution of the grid used for computing the
-            profiles
-        speed_method: speed method used to compute the
+        walkable_area: geometry for which the speed profiles are
+            computed.
+        grid_size: The resolution of the grid used for computing the
+            profiles, expressed in the same units as the `walkable_area`.
+        speed_method: The speed method used to compute the
             speed profile
-        grid_intersections_area: intersection of grid cells with the Voronoi
+        grid_intersections_area: (Optional) intersection areas of grid cells
+            with Voronoi polygons, required for some speed methods.
             polygons (result from
```

Comment:
The text seems to be messed up here a bit, due to merging?

Reply:
<img width="692" alt="Screenshot 2024-04-04 at 07 18 36" src="https://github.com/PedestrianDynamics/PedPy/assets/5772973/89c37012-baab-40ef-bf6c-e848135de69a">

what seems messed up?

Reply:
The text itself:
intersection areas of grid cells  with Voronoi polygons, required for some speed methods. polygons (result from

The sentence starting with "polygons" seems to be duplicated

Reply:
Good catch. Thanks.